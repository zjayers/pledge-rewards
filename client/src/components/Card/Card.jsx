/* Import Modules */
import React, { useState } from 'react';
/* Import Components */
import Container from '../Shared/Container/Container';
import Overlay from '../Shared/Overlay/Overlay';
import Heading from '../Shared/Heading/Heading';
import SubHeading from '../Shared/SubHeading/SubHeading';
import Title from '../Shared/Title/Title';
import Description from '../Shared/Description/Description';
import RewardList from '../RewardList/RewardList';
import SplitContainer from '../Shared/SplitContainer/SplitContainer';
import SmallTitle from '../Shared/SmallTitle/SmallTitle';
import BackerHeading from '../Shared/BackerHeading/BackerHeading';
import PledgeForm from '../PledgeForm/PledgeForm';

/**
 * Card Component
 * @param props
 * @returns {*}
 * @constructor
 */
const Card = (props) => {
  /* Use Hooks to create state component for the description container */
  const [descOpen, setDescOpen] = useState(false);
  const [activated, setActivated] = useState(false);

  const { reward } = props;
  const {
    id,
    title,
    pledgeAmount,
    description,
    deliveryMonth,
    deliveryYear,
    rewardQuantity,
    rewardItems,
  } = reward;

  const currencyConverted = Math.floor(pledgeAmount * 0.8);

  /* Return the JSX to render */
  return (
    <Container activated={activated}>
      <Overlay activated={activated} setActivated={setActivated} />
      <Heading>{`Pledge $${pledgeAmount} or more`}</Heading>
      <SubHeading uppercase>{`About Â£${currencyConverted}`}</SubHeading>
      <Title uppercase>{title}</Title>
      <Description
        descOpen={descOpen}
        activated={activated}
        setDescOpen={setDescOpen}
        description={description}
      />
      <RewardList id={id} activated={activated} rewardItems={rewardItems} />
      <SplitContainer>
        <div>
          <SubHeading uppercase>
            Estimated
            <br />
            delivery
          </SubHeading>
          <SmallTitle uppercase={false}>
            {`${deliveryMonth.substring(0, 3)} ${deliveryYear}`}
          </SmallTitle>
        </div>
        <div>
          <SubHeading uppercase>Ships To</SubHeading>
          <SmallTitle uppercase={false}>Anywhere in the world</SmallTitle>
        </div>
      </SplitContainer>
      <BackerHeading activated={activated}>
        {`${rewardQuantity} Backers`}
      </BackerHeading>
      <PledgeForm activated={activated} />
    </Container>
  );
};

/* Export the component - use the withTheme hook to allow theme access inside of this component */
export default Card;
