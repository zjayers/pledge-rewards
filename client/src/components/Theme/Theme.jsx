import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
