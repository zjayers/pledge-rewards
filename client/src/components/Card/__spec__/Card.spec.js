import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from '../Card';
import Theme from '../../Theme/Theme';

describe('Card Component', () => {
  let component;

  const reward = {
    id: 1,
    title: 'Placeholder',
    pledgeAmount: 'Placeholder',
    description: 'Placeholder',
    deliveryMonth: 'Placeholder',
    deliveryYear: 2020,
    rewardQuantity: 1,
    rewardItems: 'Test1,Test2,Test3',
  };
  beforeEach(() => {
    component = mount(
      <Theme>
        <Card reward={reward} />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
