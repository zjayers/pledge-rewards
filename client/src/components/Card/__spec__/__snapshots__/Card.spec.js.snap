// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <Card
      reward={
        Object {
          "deliveryMonth": "Placeholder",
          "deliveryYear": 2020,
          "description": "Placeholder",
          "id": 1,
          "pledgeAmount": "Placeholder",
          "rewardItems": "Test1,Test2,Test3",
          "rewardQuantity": 1,
          "title": "Placeholder",
        }
      }
    >
      <_default
        activated={false}
      >
        <Containerstyle___default
          activated={false}
        >
          <div
            className="Containerstyle___default-sc-1q8wjhx-0 BmWDh"
          >
            <_default
              activated={false}
              setActivated={[Function]}
            >
              <Overlaystyle___default
                onClick={[Function]}
              >
                <div
                  className="Overlaystyle___default-trsgfk-0 fGHlti"
                  onClick={[Function]}
                >
                  <p>
                    Select this reward
                  </p>
                </div>
              </Overlaystyle___default>
            </_default>
            <_default>
              <Headingstyle___default>
                <h2
                  className="Headingstyle___default-dlpxpq-0 iErRtB"
                >
                  Pledge $Placeholder or more
                </h2>
              </Headingstyle___default>
            </_default>
            <_default
              uppercase={true}
            >
              <SubHeadingstyle___default
                uppercase={true}
              >
                <p
                  className="SubHeadingstyle___default-qgikex-0 cDJTT"
                >
                  About Â£NaN
                </p>
              </SubHeadingstyle___default>
            </_default>
            <_default
              uppercase={true}
            >
              <Titlestyle___default
                uppercase={true}
              >
                <h3
                  className="Titlestyle___default-s64se5-0 hzYnKD"
                >
                  Placeholder
                </h3>
              </Titlestyle___default>
            </_default>
            <Description
              activated={false}
              descOpen={false}
              description="Placeholder"
              setDescOpen={[Function]}
            >
              <Descriptionstyle__DescriptionContainer
                activated={false}
                open={false}
              >
                <div
                  className="Descriptionstyle__DescriptionContainer-wtqy6i-0 lhnXnZ"
                  open={false}
                >
                  <p>
                    Placeholder
                  </p>
                  <Descriptionstyle__DescCollapseBtn
                    onClick={[Function]}
                    open={false}
                  >
                    <p
                      className="Descriptionstyle__DescCollapseBtn-wtqy6i-1 loBJcu"
                      onClick={[Function]}
                      open={false}
                    />
                  </Descriptionstyle__DescCollapseBtn>
                </div>
              </Descriptionstyle__DescriptionContainer>
            </Description>
            <_default
              activated={false}
              id={1}
              rewardItems="Test1,Test2,Test3"
            >
              <_default
                uppercase={true}
              >
                <SubHeadingstyle___default
                  uppercase={true}
                >
                  <p
                    className="SubHeadingstyle___default-qgikex-0 cDJTT"
                  >
                    Includes:
                  </p>
                </SubHeadingstyle___default>
              </_default>
              <RewarsListstyle___default>
                <ul
                  className="RewarsListstyle___default-gbe8yx-0 gRmdJF"
                >
                  <li
                    key="1_Test1-0"
                  >
                    Test1
                  </li>
                  <li
                    key="1_Test2-1"
                  >
                    Test2
                  </li>
                  <li
                    key="1_Test3-2"
                  >
                    Test3
                  </li>
                </ul>
              </RewarsListstyle___default>
            </_default>
            <_default>
              <SplitContainerstyle___default>
                <div
                  className="SplitContainerstyle___default-sc-1ep4whv-0 hdmVpp"
                >
                  <div>
                    <_default
                      uppercase={true}
                    >
                      <SubHeadingstyle___default
                        uppercase={true}
                      >
                        <p
                          className="SubHeadingstyle___default-qgikex-0 cDJTT"
                        >
                          Estimated
                          <br />
                          delivery
                        </p>
                      </SubHeadingstyle___default>
                    </_default>
                    <_default
                      uppercase={false}
                    >
                      <SmallTitlestyle___default
                        uppercase={false}
                      >
                        <h3
                          className="SmallTitlestyle___default-g42rlo-0 hVAPix"
                        >
                          Pla 2020
                        </h3>
                      </SmallTitlestyle___default>
                    </_default>
                  </div>
                  <div>
                    <_default
                      uppercase={true}
                    >
                      <SubHeadingstyle___default
                        uppercase={true}
                      >
                        <p
                          className="SubHeadingstyle___default-qgikex-0 cDJTT"
                        >
                          Ships To
                        </p>
                      </SubHeadingstyle___default>
                    </_default>
                    <_default
                      uppercase={false}
                    >
                      <SmallTitlestyle___default
                        uppercase={false}
                      >
                        <h3
                          className="SmallTitlestyle___default-g42rlo-0 hVAPix"
                        >
                          Anywhere in the world
                        </h3>
                      </SmallTitlestyle___default>
                    </_default>
                  </div>
                </div>
              </SplitContainerstyle___default>
            </_default>
            <_default
              activated={false}
            >
              <BackerHeadingstyle___default
                activated={false}
              >
                <p
                  className="BackerHeadingstyle___default-t7usxk-0 lhsgcG"
                >
                  1 Backers
                </p>
              </BackerHeadingstyle___default>
            </_default>
            <WithTheme(Component)
              activated={false}
            >
              <Component
                activated={false}
                theme={
                  Object {
                    "colors": Object {
                      "borderGrey": "#9B9E9E",
                      "buttonGreen": "#009E74",
                      "flavorText": "#888B8B",
                      "gradientBlue": "#DBE7FE",
                      "gradientOrange": "#FFF2EC",
                      "highlightGreen": "#009E74",
                      "linkText": "#498F80",
                      "text": "#282828",
                    },
                    "fontSizes": Object {
                      "large": "3rem",
                      "medium": "1.8rem",
                      "small": "1rem",
                    },
                    "fonts": Array [
                      "Muli",
                      "sans-serif",
                    ],
                  }
                }
              />
            </WithTheme(Component)>
          </div>
        </Containerstyle___default>
      </_default>
    </Card>
  </ThemeProvider>
</Theme>
`;
